crates/package-manager/src/install_without_lockfile.rs

```diff
@@ -75,7 +81,18 @@ impl<'a> InstallWithoutLockfile<'a, ()> {
     /// Install dependencies of a dependency.
     #[async_recursion]
     async fn install_dependencies_from_registry(&self, package: &PackageVersion) {
-        let InstallWithoutLockfile { tarball_mem_cache, http_client, config, .. } = self;
+        let InstallWithoutLockfile {
+            tarball_mem_cache,
+            http_client,
+            config,
+            resolved_packages,
+            ..
+        } = self;
+
+        if !resolved_packages.insert(package.to_virtual_store_name(), true) {
```

Comment:
There's no `false`? Then I think `HashSet` might be better fit.

Reply:
Thanks for review, change it to dashset now.